From 818ef6ae5a891574e803e5db32794b7a57927cdf Mon Sep 17 00:00:00 2001
From: Tzvetomir Stoyanov <tz.stoyanov@gmail.com>
Date: Thu, 9 Jul 2026 19:55:45 +0300
Subject: [PATCH] Fix linker issue with linker_definitions.ld

The linker_definitions.ld is generated in CMAKE_BINARY_DIR location,
which is the build root of the project. However, the linker searches
for it in CMAKE_CURRENT_BINARY_DIR location, which is the build
directory of the fota bootloader.

Signed-off-by: Tzvetomir Stoyanov <tz.stoyanov@gmail.com
---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libs/pico_fota_bootloader/CMakeLists.txt b/libs/pico_fota_bootloader/CMakeLists.txt
index 3bd514b..4094de2 100644
--- a/libs/pico_fota_bootloader/CMakeLists.txt
+++ b/libs/pico_fota_bootloader/CMakeLists.txt
@@ -99,8 +99,9 @@ target_link_libraries(pico_fota_bootloader_lib PUBLIC
                       hardware_flash)
 target_link_libraries(pico_fota_bootloader_lib PRIVATE
                       pico_mbedtls)
+target_link_directories(pico_fota_bootloader_lib PUBLIC "${CMAKE_BINARY_DIR}")
 target_link_options(pico_fota_bootloader_lib PRIVATE
-                    "-T${CMAKE_CURRENT_BINARY_DIR}/linker_definitions.ld")
+                    "-T${CMAKE_BINARY_DIR}/linker_definitions.ld")
 target_compile_options(pico_fota_bootloader_lib PRIVATE
                        -Wall
                        -Wextra
-- 
2.54.0

